src/video/cameradevice.cpp

```diff
@@ -225,7 +225,6 @@ QVector<QPair<QString, QString>> CameraDevice::getRawDeviceListGeneric()
     if (av_opt_set_dict2(s, &tmp, AV_OPT_SEARCH_CHILDREN) < 0)
     {
         av_dict_free(&tmp);
-        avformat_free_context(s);
```

Comment:
You're introducing a new leak here if we return early at line 233
Instead, if av_opt_set_dict2 fails we should keep the avformat_free_context(s); at line 228 and return early here, and if it doesn't fail we should add:

````
av_dict_free(&tmp);
avformat_free_context(s);
```

right after the avdevice_list_devices (no need to wait until the end of the function)